airbyte-cdk/python/unit_tests/sources/streams/test_stream_read.py

```diff
@@ -287,6 +301,10 @@ def test_full_refresh_read_two_slices(constructor):
 
     actual_records = _read(stream, configured_stream, logger, slice_logger, message_repository, state_manager, internal_config)
 
+    if constructor == _concurrent_stream:
+        assert hasattr(stream._cursor, "state")
+        assert str(stream._cursor.state) == "{'__ab_full_refresh_state_message': True}"
```

Comment:
Weirdly enough you can even check this in the beginning before the `_read` because the cursor has a constant state. but we can check it here at the end

Later when we switch to incremental checkpointing as it wants, if we give it the right cursor, we should get proper state messages with the right cursor